plugins/catalog/src/js/repositories/RepositoriesAdd.js

```diff
@@ -32,8 +32,10 @@ const getErrorMessage = (response = {}) => {
 
 const addPackageRepositoryMutation = gql`
   mutation {
-    addPackageRepository(name: $name, uri: $uri, priority: $index) {
+    addPackageRepository(name: $name, uri: $uri, index: $index) {
       name
+      uri
```

Comment:
so we haven't been passing `uri` as well?

Reply:
Apparently not.

Reply:
We were passing the `uri`  that is done with the context parameter `$uri` in the mutation call. Adding uri & index on lines 37 & 38 ensures those fields are returned on the result.